terraform-development-factory/main.tf

```
@@ -0,0 +1,149 @@
+module "organization" {
+  source = "github.com/harness-community/terraform-harness-structure//modules/organizations"
+
+  name     = var.organization_name
+  existing = var.create_organization ? false : true
+}
+
+module "project" {
+  source = "github.com/harness-community/terraform-harness-structure//modules/projects"
+
+  name            = var.project_name
+  organization_id = module.organization.organization_details.id
+}
+
+module "templates" {
+  source = "git@github.com:harness-community/terraform-harness-content.git//modules/templates"
+
+  name            = "Terraform Validation"
+  organization_id = module.organization.organization_details.id
+  project_id      = module.project.project_details.id
+  yaml_data = templatefile(
+    "templates/templates/terraform-deployment.yaml",
+    {
+      TERRAFORM_FILES_PATH : var.terraform_files
+      MAX_CONCURRENCY : var.max_concurrency
+    }
+  )
+  template_version = "v1.0.0"
+  type             = "Pipeline"
+  tags = {
+    role = "Terraform Validation"
+  }
+
+}
+
+module "pipelines" {
+  source = "git@github.com:harness-community/terraform-harness-content.git//modules/pipelines"
+  for_each = {
+    for pipeline in var.repositories : (pipeline) => pipeline
+  }
+
+  name            = split("/", each.value)[1]
```

```suggestion
  name            = element(split("/", each.value), length(split("/", each.value)) - 1)
```

this is written in a way that requires the github connector to be a github-wide connector. Lots of people use org based connectors, which only take in the repo name and not the org. this was causing the trigger to break by using `rssnyder/terraform` for the repo name rather than `terraform`.

doing this change here lets me set `terraform` as var.repositories and not `rssnyder/terraform`.

---

I see your point and that makes sense.  Pushed the update